### PR TITLE
task/WMAQA-38 - Refactor tests to be portal agnostic

### PR DESCRIPTION
**Pull request by fnets**
Tests should now work with Frontera Portal, and skip tests if the portal has those applications disabled.

To test, run the full suite of tests against the existing CEP configuration.  Then:

1. Replace `settings/custom_portal_settings.py` with the one for [frontera](https://github.com/TACC/Core-Portal-Deployments/blob/main/frontera-portal/camino/prod.settings_custom.py). 
2. Replace the `NGINX_SERVER_NAME` in `.env.portal` with `frontera-portal.tacc.utexas.edu`
3. Run `python3 pythonHelper.py` and make sure that the JSON is created. 
4. Run the full test suite, it should now run against the frontera portal.

### CODE REVIEW

File: tests/publicData/publicData.spec.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-import { test } from '../../fixtures/baseFixture'
- 
-
+import { test } from '../../fixtures/baseFixture';
+import { NGINX_SERVER_NAME } from '../../settings/custom_portal_settings.json';
 
 test.describe('Public Data Tests', () => {
-  
+  test.skip(NGINX_SERVER_NAME != 'cep.tacc.utexas.edu', 'Public Data not on portal, test skipped');
+
```
Comment (shayanaijaz) [inline tests/publicData/publicData.spec.js:7]: There are other portals such as [3dem](https://3dem.org/) that also have Public Data access. Might need a different approach to filter out systems with Public Data

**Review by shayanaijaz — CHANGES_REQUESTED** [review]
In prod frontera, I have one test failing which is [unauthorized-user.spec.js](https://github.com/TACC/Core-Portal-E2E/blob/main/tests/unauthorized-user/unauthorized-user.spec.js). It is failing because that test also checks Public Data and there isn't a skip condition added in there. 

I also ran the tests on pprd frontera (which we should also test and add support for in this PR) and that one seems to be failing on the Onboarding Admin tests since the user does not have admin access there. We should probably add a condition to skip the Onboarding Admin tests if the user doesn't have admin. 

One way we can check for admin status is by visiting the django core admin site (e.g [frontera-portal.tacc.utexas.edu/core/admin](frontera-portal.tacc.utexas.edu/core/admin)). If the user has admin access they should be able to view the Site Administration page. If they don't, then they are redirected to the login page and shown an error message. We can use this to determine admin status, and skip Onboarding Admin tests if they are shown the error message. Let me know what you think

File: tests/publicData/publicData.spec.js (hunk 1 above)
Comment (fnets) [reply tests/publicData/publicData.spec.js:7]: Good catch, it's always the portal you forget to check. I think I have a way around this, by checking Storage Systems for Public Data. But, that doesn't guarantee that there will be a CMS button available (MISE, for example).

**Comment by fnets** [comment]
> In prod frontera, I have one test failing which is [unauthorized-user.spec.js](https://github.com/TACC/Core-Portal-E2E/blob/main/tests/unauthorized-user/unauthorized-user.spec.js). It is failing because that test also checks Public Data and there isn't a skip condition added in there.

Good catch! Don't know how I missed that.

>Pprd frontera (which we should also test and add support for in this PR) and that one seems to be failing on the Onboarding Admin tests since the user does not have admin access there. We should probably add a condition to skip the Onboarding Admin tests if the user doesn't have admin.

I think your suggested test is a good one, but I can't decide if its more important to test that the Onboarding works or more important to test when it shouldn't work.

**Review by shayanaijaz — APPROVED** [review]
LGTM. All tests pass other than the compress and extract apps which I'll take a look at

**Pull request merged**